### PR TITLE
fix(sweeps): unbork launch specs with placeholder uris for sweep schedulers

### PR DESCRIPTION
**Pull request by bcsherma**
Description
-----------

- Fixes WB-NNNNN
- Fixes #NNNN

What does the PR do? Include a concise description of the PR contents.

- [ ] I updated CHANGELOG.md, or it's not applicable

Testing
-------
How was this PR tested?

### CODE REVIEW

**Comment by bcsherma** [comment]
* **#7539** <a href="https://app.graphite.dev/github/pr/wandb/wandb/7539?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bcsherma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: wandb/sdk/launch/_project_spec.py [hunk 1]
```diff
@@ -139,8 +141,10 @@ def init_source(self) -> None:
         elif self.job is not None:
             self.source = LaunchSource.JOB
             self.project_dir = tempfile.mkdtemp()
-        else:
-            raise LaunchError("Launch project must have a source.")
+        if self.uri and self.uri.startswith("placeholder"):
+            self.source = LaunchSource.SCHEDULER
+            self.project_dir = os.getcwd()
+            self._entry_point = self.override_entrypoint
```
Comment (KyleGoyette) [inline wandb/sdk/launch/_project_spec.py:147]: Why set the entrypoint to override entrypoint?

Comment (bcsherma) [reply wandb/sdk/launch/_project_spec.py:147]: I was trying to accommodate [this](https://github.com/wandb/wandb/blob/531d06c2db15b50220c36dc807cff9734e3d9a60/wandb/sdk/launch/_project_spec.py#L341)

We could instead modify that error case to permit `LaunchSource.Scheduler`

**Review by KyleGoyette — APPROVED** [review]
LGTM

**Pull request merged**